src/utils.rs

```diff
@@ -48,43 +48,34 @@ pub(crate) fn epoch_ms_to_ms_in_day(t: i64) -> u32 {
     (t.rem_euclid(i64::from(MS_PER_DAY))) as u32
 }
 
-/// Mathematically determine the days in a year.
-pub(crate) fn mathematical_days_in_year(y: i32) -> i32 {
+pub(crate) fn is_leap(y: i32) -> bool {
```

Comment:
note: the slightly odd util names comes from the Temporal specification, mainly [Section 13.3](https://tc39.es/proposal-temporal/#sec-date-equations). I'm not against cleaning the names, but it may be worth commenting the date equation these functions are associated with.